re.c

```diff
@@ -3853,6 +3853,10 @@ reg_copy(VALUE copy, VALUE orig)
     RB_OBJ_WRITE(copy, &RREGEXP(copy)->src, RREGEXP(orig)->src);
     RREGEXP_PTR(copy)->timelimit = RREGEXP_PTR(orig)->timelimit;
     rb_enc_copy(copy, orig);
+
+    if (RB_FL_TEST(orig, KCODE_FIXED)) RB_FL_SET(copy, KCODE_FIXED);
+    if (RB_FL_TEST(orig, REG_ENCODING_NONE)) RB_FL_SET(copy, REG_ENCODING_NONE);
```

Comment:
I prefer the way I have it since it is more readable to me and better follows how these are set during initialization ([ref](https://github.com/ruby/ruby/blob/master/re.c#L3258-L3263)). There wasn't a performance difference between the two that I could tell using the benchmarking tools.